File: Front/EInmobiliaria/app/src/main/java/com/example/einmobiliaria/models/in/ElementModel.kt
package com.example.einmobiliaria.models.`in`

import com.google.gson.annotations.SerializedName

data class ElementModel(

    @SerializedName("name")
    val Name: String,
    @SerializedName("amount")
    val Amount: Int,
    @SerializedName("isBroken")
    val IsBroken: Boolean,
    @SerializedName("photoName")
    val PhotoName: String,
    @SerializedName("image")
    val Image: Int,
    @SerializedName("apartmentId")
    val ApartmentId: Int,
    @SerializedName("userId")
    val UserId: Int,
)

File: BackendApp/Model/In/ElementModel.cs
using System;
using Domain.Entities;

namespace Model.In
{
    public class ElementModel
    {
        public string Name { get; set; }
        public int Amount { get; set; }
        public bool IsBroken { get; set; }
        public string PhotoName { get; set; }   
        public int Image {get; set;} 
        public int ApartmentId { get; set; }
        public int UserId { get; set; } 
        public Element ToEntity(bool post = true)
        {
            if (post && this.AnyValueEmpty())
            {
                throw new ArgumentException("Se necesita nombre y cantidad");
            }
            Element elementModel =  new Element()
            {
    
                Name = this.Name,
                Amount = this.Amount,
                IsBroken = this.IsBroken,
                ApartmentId = this.ApartmentId,
                UserId = this.UserId
            };
            if (elementModel.IsBroken)
            {
                elementModel.Photo = new Photo();
                elementModel.Photo.Name = this.PhotoName;
                elementModel.Photo.Image = this.Image;
            }
            return elementModel;
        }
        public bool AnyValueEmpty()
        {
            bool nameEmpty = this.Name == "";
            bool amountEmpty = this.Amount < 1;
            bool isBroken = this.IsBroken && this.PhotoName == "";
            return nameEmpty || amountEmpty  || isBroken;
        }   
    }
}